### PR TITLE
Improve remove query strings

### PR DESCRIPTION
**Pull request by Tabrisrp**
* Use filesystem when possible
* Reproduce original tree to file
* Prevent issue with relative paths

### CODE REVIEW

File: inc/functions/formatting.php [hunk 1]
```diff
@@ -231,3 +231,31 @@ function rocket_get_cache_busting_paths( $filename, $extension ) {
...
+ 
+	foreach ( explode( '/', $file_path ) as $part ) {
+		if ( empty( $part ) || '.' === $part ) {
+			continue;
+		}
+
+		if ( '..' !== $part ) {
+			array_push( $path, $part );
+		}
+		else if ( count( $path ) > 0 ) {
```
Comment (Screenfeed) [inline inc/functions/formatting.php]: `elseif`

File: inc/functions/formatting.php [hunk 2]
```diff
@@ -231,3 +231,31 @@ function rocket_get_cache_busting_paths( $filename, $extension ) {
...
+ * @author Remy Perona
+ *
+ * @param string $file_path Path to the file.
+ * @return string Resolved file path
+ */
+function rocket_realpath( $file_path ) {
+	$path = [];
+ 
+	foreach ( explode( '/', $file_path ) as $part ) {
+		if ( empty( $part ) || '.' === $part ) {
```
Comment (Screenfeed) [inline inc/functions/formatting.php]:
```var_dump( rocket_realpath(  '/foo/bar/0/baz/' ) );
 string(11) 'foo/bar/baz'```
I'm not sure this what you expect.

File: inc/functions/files.php [hunk 3]
```diff
@@ -397,23 +397,30 @@ function rocket_clean_minify( $extensions = array( 'js', 'css' ) ) {
...
 		 * @param string $ext File extensions to minify.
 		*/
-		do_action( 'before_rocket_clean_cache_busting', $ext );
+		do_action( 'before_rocket_clean_minify', $ext );
 
-		$files = @glob( WP_ROCKET_CACHE_BUSTING_PATH . $blog_id . '/*.' . $ext, GLOB_NOSORT );
-		if ( $files ) {
-			foreach ( $files as $file ) { // no array map to use @.
-				rocket_direct_filesystem()->delete( $file );
+		try {
```
Comment (Tabrisrp) [inline inc/functions/files.php]: @Screenfeed could use your insight on optimizing this part if you see anything since there is multiple `foreach`

Comment (Screenfeed) [reply inc/functions/files.php]: @Tabrisrp I'm not familiar with iterators at all but I have an idea, I must do some tests (use a filter after `new RecursiveDirectoryIterator()`).

**Pull request merged**